k-distribution/include/builtin/domains.k

```diff
@@ -980,6 +987,10 @@ module K-IO
                  | "#systemResult" "(" Int /* exit code */ "," String /* stdout */ "," String /* stderr */ ")"
 
   syntax K ::= #logToFile(String, String) [function, functional, hook(IO.log), impure, returnsUnit]
+  // #tempFilename ( prefix:String, suffix:String):IOString - creates a temporary file in the default system location
+  syntax IOString ::= #tempFilename(String, String)  [function, functional, hook(IO.tempFilename), impure]
+  syntax KItem ::= #remove(String) [function, functional, hook(IO.remove), impure, returnsUnit]
```

Comment:
This should return a K, not a KItem, since its default return value is .K. Note that it needs to return an IOError upon failure and as such it is not sound to possess the returnsUnit attribute.